heroic-core/src/main/java/com/spotify/heroic/http/tracing/TextFormatGetter.java

```diff
@@ -35,6 +35,10 @@ public String get(final Object o, final String s) {
     final ContainerRequest request = (ContainerRequest) o;
     final List<String> requestHeader = request.getRequestHeader(s);
 
+    if (requestHeader == null) {
+      return null;
+    }
+
     if (requestHeader.size() > 0) {
```

Comment:
You could combine the two checks, `if (requestHeader != null && requestHeader.size() > 0)`